gobblin-runtime/src/main/java/gobblin/runtime/Fork.java

```diff
@@ -60,7 +64,7 @@
  * @author ynli
  */
 @SuppressWarnings("unchecked")
-public class Fork implements Closeable, Runnable, FinalState {
+public class Fork extends Instrumented implements Closeable, Runnable, FinalState {
```

Comment:
I hold a reference to `MetricContext` and close it in the `Fork.close()` method.